### ISSUE TITLE
Double clicking a .tx file should open it in Tiled.app

### ISSUE THREAD

**Issue by hotdogsoup-nl** [mapeditor/tiled#3372]
**Describe the bug**
Double clicking a .tx file should open it in Tiled.app

**To Reproduce**
1. open a Finder window. 
2. double click on a .tx file

***Expected behavior**
Tiled.app should open the template file in the Template editor.

**Work around**
By holding Option+CMD the .tx file can be dragged & dropped onto the Tiled.app icon. Now the file is opened correctly.

**Specifications:**
 - macOS 12.3.1
 - Tiled Version: 1.8.5

**Comment by bjorn** [comment]
Yeah, that's a good thing to look into. I'm not sure if it's enough to just add a line with `tx` to [the Info.plist file](https://github.com/mapeditor/tiled/blob/7f7589e54d824fc28949f5ad914a8e5b2377f7cd/src/tiledapp/Info.plist#L22), but that'd be a start at least. Probably `tsx`, as well as `tmj`, `tsj` and `tj` (more specific extensions supported by the JSON format) should be added there as well.

**Comment by eishiya** [comment]
It would be nice if the Windows installer also associated those other formats with Tiled, ideally optionally. Looks like only TMX is registered:
https://github.com/mapeditor/tiled/blob/3d8724f91ba53d7c1d245b5a8f778662a7aa5d0d/dist/win/installer.wxs#L142-L147

It looks like in some versions of Windows (or at least in some combinations of settings?), manual registrations display a different, less clear icon, so even though it's fairly easy to make the associations manually after installing, it would be nice to have the installer make them.
Screenshot from Mitsuru Tenma on Discord: https://discord.com/channels/524610627545595904/524610627545595906/981968829473239040 (I'll edit this post with a copy of the image if I get their permission to do so.)
On my Windows manual associations look fine, but I can see why they'd be suboptimal on theirs. I couldn't replicate their icon appearance via various settings, so it may be a version difference, in which case, the ugliness is there to stay xP

**Comment by bjorn** [comment]
Implemented for macOS by #4469. Windows installer additions still pending...
